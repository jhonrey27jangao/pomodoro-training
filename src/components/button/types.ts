export type ButtonProps = {
  type?: string;
  rounded?: boolean;
  theme?: "success" | "warning" | "default" | "dark";
  width?: number;
  center?: boolean;
  children?: any;
  onClick?:
    | ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
    | undefined;
  disabled?: boolean;
};
