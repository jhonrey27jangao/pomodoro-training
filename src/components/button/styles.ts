import styled from "styled-components";
import { ButtonProps } from "./types";

export const SpanBtn = styled.span`
  position: relative;
  z-index: 1;
`;

export const StyledNonDefaultBtn = styled.button<ButtonProps>`
    margin: 5px auto;
    width: ${(props) => (props.width ? `${props.width}px` : "100%")};
    border: none;
    border-radius: 30px;
    display: block;
    text-align: center;
    cursor: pointer;
    text-transform: uppercase;
    outline: none;
    overflow: hidden;
    position: relative;
    color: #fff;
    font-weight: 700;
    font-size: 13px;
    background-color: ${(props) =>
      props.theme === "success" ? "#222;" : "#7e0109;"}
    padding: 5px 15px;
    box-shadow: 0 5px 15px rgba(0,0,0,0.20);

    &:disabled {
      opacity: 0.8;
    }

    &:after {
        content: "";
        position: absolute;
        left: 0;
        top: 0;
        height: 490%;
        width: 140%;
        background: #111;
        -webkit-transition: all .5s ease-in-out;
        transition: all .5s ease-in-out;
        -webkit-transform: translateX(-98%) translateY(-25%) rotate(45deg);
        transform: translateX(-98%) translateY(-25%) rotate(45deg);
        background-color: none;
        opacity: 0;
    }
    &:hover:after {
        -webkit-transform: translateX(-9%) translateY(-25%) rotate(45deg);
        transform: translateX(-9%) translateY(-25%) rotate(45deg);
        opacity: 1;
    }`;

export const StyledDefaultButton = styled.button<ButtonProps>`
  margin: 10px auto;
  width: ${(props) => (props.width ? `${props.width}px` : "100%")};
  border: ${(props) =>
    props.theme === "default" ? `3px solid #000` : "3px solid #fff"};
  text-align: center;
  cursor: pointer;
  text-transform: uppercase;
  outline: none;
  color: ${(props) => (props.theme === "default" ? `#000` : "#fff")};
  font-weight: 700;
  font-size: 13px;
  background-color: ${(props) =>
    props.theme === "default"
      ? "#fff"
      : props.theme === "dark"
      ? "#000"
      : "#7e0109"};
  padding: 20px 15px;
  border-radius: ${(props) => (props.rounded ? `100%` : "0")};
  &:hover {
    filter: invert();
  }
`;
