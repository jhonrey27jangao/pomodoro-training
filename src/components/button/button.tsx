import React from "react";
import { ButtonProps } from "./types";
import { SpanBtn, StyledDefaultButton, StyledNonDefaultBtn } from "./styles";

export const Button: React.FC<ButtonProps> = ({
  width,
  theme,
  onClick,
  children,
  disabled,
  rounded,
}) => {
  return (
    <>
      {" "}
      {theme === "dark" ? (
        <>
          <StyledDefaultButton
            width={width}
            theme={theme}
            onClick={onClick}
            disabled={disabled}
            rounded={rounded}
          >
            <SpanBtn>{children}</SpanBtn>
          </StyledDefaultButton>
        </>
      ) : theme === "default" ? (
        <>
          <StyledDefaultButton
            width={width}
            theme={theme}
            onClick={onClick}
            disabled={disabled}
            rounded={rounded}
          >
            <SpanBtn>{children}</SpanBtn>
          </StyledDefaultButton>
        </>
      ) : theme === "success" ? (
        <>
          <StyledNonDefaultBtn
            width={width}
            theme={theme}
            onClick={onClick}
            disabled={disabled}
          >
            {children}
          </StyledNonDefaultBtn>
        </>
      ) : (
        <>
          <StyledNonDefaultBtn
            width={width}
            theme={theme}
            onClick={onClick}
            disabled={disabled}
          >
            {children}
          </StyledNonDefaultBtn>
        </>
      )}
    </>
  );
};
