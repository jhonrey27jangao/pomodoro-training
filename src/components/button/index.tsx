import React from "react";
import { ButtonProps } from "./types";
import { SpanBtn, StyledBtn } from "./styles";

export const Button: React.FC<ButtonProps> = ({
  width,
  theme,
  onClick,
  children,
  disabled
}) => {
  return (
    <>
      <StyledBtn width={width} theme={theme} onClick={onClick} disabled={disabled}>
        <SpanBtn>{children}</SpanBtn>
      </StyledBtn>
    </>
  );
};
