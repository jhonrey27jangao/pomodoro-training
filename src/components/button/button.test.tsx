import React from "react";
import { render } from "@testing-library/react";
import { Button } from "./button";

describe("Test for Button Component", () => {
  it("should render Button with dark theme", () => {
    const { container } = render(<Button theme="dark" />);
    expect(container).toMatchSnapshot();
  });

  it("should render Button  default theme", () => {
    const { container } = render(<Button theme="default" />);
    expect(container).toMatchSnapshot();
  });

  it("should render Button  success theme", () => {
    const { container } = render(<Button theme="success" />);
    expect(container).toMatchSnapshot();
  });

  it("should render Button  warning theme", () => {
    const { container } = render(<Button theme="warning" />);
    expect(container).toMatchSnapshot();
  });
});
