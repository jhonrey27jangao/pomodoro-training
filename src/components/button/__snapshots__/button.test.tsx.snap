// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Button Component should render Button  default theme 1`] = `
<div>
  <button
    class="sc-AxiKw jXfssU"
    type="submit"
  >
    <span
      class="sc-AxjAm eliavK"
    />
  </button>
</div>
`;

exports[`Test for Button Component should render Button  success theme 1`] = `
<div>
  <button
    class="sc-AxirZ Vhqeo"
    type="submit"
  />
</div>
`;

exports[`Test for Button Component should render Button  warning theme 1`] = `
<div>
  <button
    class="sc-AxirZ bULvc"
    type="submit"
  />
</div>
`;

exports[`Test for Button Component should render Button with dark theme 1`] = `
<div>
  <button
    class="sc-AxiKw dOcvua"
    type="submit"
  >
    <span
      class="sc-AxjAm eliavK"
    />
  </button>
</div>
`;
